Settings.c

```diff
@@ -169,10 +169,10 @@ static bool Settings_read(Settings* this, const char* fileName, unsigned int ini
       if (String_eq(option[0], "config_reader_min_version")) {
          this->config_version = atoi(option[1]);
          if (this->config_version > CONFIG_READER_MIN_VERSION) {
-             // the version of the config file on disk is newer than what we can read
-             fprintf(stderr, "WARNING: The config file %s requires support for a newer format (v%d) than what this version of htop can read (v%d).\n", fileName, this->config_version, CONFIG_READER_MIN_VERSION);
-             fprintf(stderr, "         It will be overwritten when this version of htop exits.\n");
-             return false;
+            // the version of the config file on disk is newer than what we can read
+            fprintf(stderr, "WARNING: %s specifies configuration format version v%d, but this %s binary supports up to v%d.", fileName, this->config_version, PACKAGE, CONFIG_READER_MIN_VERSION);
+            fprintf(stderr, "         The configuration version will be downgraded to v%d when %s exits.\n", CONFIG_READER_MIN_VERSION, PACKAGE);
```

Comment:
Sorry for commenting too late, but I think it might be better if htop can stop (and quit) when the config version is found out to be too new.
Or make a backup (with filename like htoprc.backup) before writing to the file.
My idea is not to destroy the user's config file as it might be working for him. And he might not be aware that the config file is incompatible before we alert him.

Reply:
P.S. Remember that users can't see the stderr message unless htop halts or crashes, so it is more likely that config file gets downgraded/overwritten before user can see the warning and do something with it.

Reply:
Yes, and that is fine. If people run downgraded versions of htop in parallel they will be able to re-create a config file from backup or by just manually setting what they wanted again.

If you find a .backup file, do you overwrite this? Do you do .backup.YYMMDDHHSSmm and fill up the hard disk? This is a rabbit hole. This file is for consumption by `htop`. The fact that it can be edited by hand is just a convenience to the developers and power users that don't complain :-).

Reply:
@fasterit You know that's one of the reasons I was skeptical of this PR. And no, I don't think people can re-create a config file from scratch that easily.

Just to tell my personal story: I had a painful experience when Firefox (the web browser) kills off my session (with all opened tabs) when it performs an update. I have a habit of leaving many tabs open in order to maintain the information I needed for work. Although Firefox does backup sessions every time it starts (which is good), it's update system is not always reliable of keeping my sessions anyway, and occasionally I lost them. It takes the hell a lot of time to lookup my browsing history and recover which web pages I had opened before. Now I always manually backup my session before a Firefox update, just to get rid of the unreliable update process.

> If you find a .backup file, do you overwrite this? Do you do .backup.YYMMDDHHSSmm and fill up the hard disk?

Don't overwrite the backup. That's what the backup is for. And Firefox does include a date for it's session backups (for Firefox upgrades; it looks like "upgrade.jsonlz4-YYYYMMDDhhmmss").

Reply:
@Explorer09 **Firefox**: Right-click on any of the tabs, choose "Select All Tabs" from the context menu, then right-click on any tab again and select "Bookmark Tabs". This will create a folder of bookmarks for you that you can restore at any time.

**htop**: `cp ~/.config/htop/htoprc{,_$(date +%y%m%d%H%M%S)}`
Put it into your `~/.bashrc` if you want to.

Reply:
@fasterit For Firefox, No, bookmarks and tabs are different use cases and shouldn't be mixed. I consider bookmarks for long term references and tabs for short term. I may not leave a tab open forever, but I need it open when my job is half done and have to continue it tomorrow. This is what I mean by "short term" and saving the tabs as bookmarks won't fit my workflow.